### PR TITLE
UGENE-5450 Problem building tree from sample alignment on Windows

### PR DESCRIPTION
**Pull request by rasputinkirill**
I don't think this bug deserve automatic GUI test. It can be checked once manually.

### CODE REVIEW

**Review by evelinabis — DISMISSED** [review]

File: src/corelibs/U2View/src/phyltree/CreatePhyTreeDialogController.cpp [hunk 1]
```diff
@@ -171,7 +172,13 @@ bool CreatePhyTreeDialogController::checkMemory() {
 
 void CreatePhyTreeDialogController::initSaveController(const MultipleSequenceAlignmentObject *msaObject) {
     SaveDocumentControllerConfig config;
-    config.defaultFileName = GUrlUtils::getNewLocalUrlByExtention(msaObject->getDocument()->getURLString(), msaObject->getGObjectName(), ".nwk", "");
+    QString urlStr;
+    if (FileAndDirectoryUtils::isDirectoryWritable(QFileInfo(msaObject->getDocument()->getURLString()).dir().path())) {
```
Comment (yalgaer) [inline src/corelibs/U2View/src/phyltree/CreatePhyTreeDialogController.cpp]: Please cache "msaObject->getDocument()->getURLString()" into a local variable to make code more readable

**Review by yalgaer — CHANGES_REQUESTED** [review]
This scenario can be covered by UI test.

File: src/corelibs/U2View/src/phyltree/CreatePhyTreeDialogController.cpp [hunk 2]
```diff
@@ -171,7 +172,14 @@ bool CreatePhyTreeDialogController::checkMemory() {
 
 void CreatePhyTreeDialogController::initSaveController(const MultipleSequenceAlignmentObject *msaObject) {
     SaveDocumentControllerConfig config;
-    config.defaultFileName = GUrlUtils::getNewLocalUrlByExtention(msaObject->getDocument()->getURLString(), msaObject->getGObjectName(), ".nwk", "");
+    QString urlStr;
+    QString msaObjUrlString = msaObject->getDocument()->getURLString();
+    if (FileAndDirectoryUtils::isDirectoryWritable(QFileInfo(msaObjUrlString).dir().path())) {
+        urlStr = msaObject->getDocument()->getURLString();
```
Comment (evelinabis) [inline src/corelibs/U2View/src/phyltree/CreatePhyTreeDialogController.cpp]: Please replace with cached variable `msaObjUrlString`

**Review by evelinabis — DISMISSED** [review]

File: src/corelibs/U2View/src/phyltree/CreatePhyTreeDialogController.cpp [hunk 3]
```diff
@@ -171,7 +172,11 @@ bool CreatePhyTreeDialogController::checkMemory() {
 
 void CreatePhyTreeDialogController::initSaveController(const MultipleSequenceAlignmentObject *msaObject) {
     SaveDocumentControllerConfig config;
-    config.defaultFileName = GUrlUtils::getNewLocalUrlByExtention(msaObject->getDocument()->getURLString(), msaObject->getGObjectName(), ".nwk", "");
+    QString urlStr = msaObject->getDocument()->getURLString();
+    if (!FileAndDirectoryUtils::isDirectoryWritable(QFileInfo(urlStr).dir().path())) {
+        urlStr = GUrlUtils::getDefaultDataPath() + "/" + msaObject->getGObjectName();
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2View/src/phyltree/CreatePhyTreeDialogController.cpp:177]: I suggest write a message to log in this case, for example:

`coreLog.details(tr("You have no write access to the \"%1\" directory, the result will be written to \"%2\".").arg(urlStr).arg(path));`

Comment (DmitriiSukhomlinov) [reply src/corelibs/U2View/src/phyltree/CreatePhyTreeDialogController.cpp:177]: Also I think it's better to save result to "data/samples/Newick", not just "data"

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
See comments

**Review by evelinabis — COMMENTED** [review]
Please, run GUI Nightly Ubuntu test

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 4]
```diff
@@ -1427,6 +1427,7 @@ void GUITestBasePlugin::registerTests(UGUITestBase *guiTestBase) {
     REGISTER_TEST(GUITest_regression_scenarios::test_5447_1);
     REGISTER_TEST(GUITest_regression_scenarios::test_5447_2);
     REGISTER_TEST(GUITest_regression_scenarios::test_5447_3);
+    REGISTER_TEST_ONLY_WINDOWS(GUITest_regression_scenarios::test_5450);
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp:1430]: Why do you register this test on Windows only? Could you please add it to all nightly builds?

Everything else looks good.

**Review by yalgaer — CHANGES_REQUESTED** [review]

**Review by evelinabis — APPROVED** [review]

**Comment by rasputinkirill** [comment]
For now this bug isn't reproducable

**Pull request closed without merging**